src/API/Google/AdsCampaign.php

```diff
@@ -118,13 +126,26 @@ public function get_campaigns(): array {
 	 */
 	public function get_campaign( int $id ): array {
 		try {
-			$results = ( new AdsCampaignQuery() )
+			$campaign_results = ( new AdsCampaignQuery() )
+				->set_client( $this->client, $this->options->get_ads_id() )
+				->where( 'campaign.id', $id )
+				->get_results();
+
+			$campaign_criterion_results = ( new AdsCampaignCriterionQuery() )
 				->set_client( $this->client, $this->options->get_ads_id() )
 				->where( 'campaign.id', $id )
 				->get_results();
 
-			foreach ( $results->iterateAllElements() as $row ) {
-				return $this->convert_campaign( $row );
+			$converted_campaigns = [];
+
+			foreach ( $campaign_results->iterateAllElements() as $row ) {
+				$converted_campaigns[] = $this->convert_campaign( $row );
+				break;
```

Comment:
❓ Why it is necessary this break here?

Reply:
The previous code before this PR was something like:

```js
foreach ( $campaign_results->iterateAllElements() as $row ) {
  return $this->convert_campaign( $row );
```

Which means we want to get the first element of `$campaign_results`, and since `$campaign_results` is an instance of `PagedListResponse` rather than an array, we couldn't use some of the array functions to retrieve the first element.

Reply:
Maybe another way would be to move this piece of logic into a function that would take the  $campaign_results->iterateAllElements() array and return the **converted_campaigns**. You could have a second argument as a boolean flag (eg. **single** ) in order to return the first index of  **converted_campaigns** only using current() or reset().

```
foreach ( $campaign_results->iterateAllElements() as $row ) {
				$campaign                               = $this->convert_campaign( $row );
				$converted_campaigns[ $campaign['id'] ] = $campaign;
			}
```